### PR TITLE
Fix divide by zero if macro used with wrong key type

### PR DESCRIPTION
**Pull request by paul-elliott-arm**
## Description
If PSA_CIPHER_ENCRYPT_OUTPUT_SIZE was called on a non symmetric key, then a divide by zero could happen, as PSA_CIPHER_BLOCK_LENGTH will return 0 for such a key, and PSA_ROUND_UP_TO_MULTIPLE will divide by theblock length.

Found by coverity (and with apologies for making an already complicated macro even more so)

## Status
**READY**

## Requires Backporting
YES
[2.x](https://github.com/ARMmbed/mbedtls/pull/4759)

## Migrations
If there is any API change, what's the incentive and logic for it.
NO

## Todos
- [x] Tests
- [x] Backported

### CODE REVIEW

File: include/psa/crypto_sizes.h [hunk 1]
```diff
@@ -992,9 +992,11 @@
...
     (alg == PSA_ALG_CBC_PKCS7 ?                                                 \
+     (((key_type) & PSA_KEY_TYPE_CATEGORY_MASK)                                 \
+                                           == PSA_KEY_TYPE_CATEGORY_SYMMETRIC ? \
      PSA_ROUND_UP_TO_MULTIPLE(PSA_BLOCK_CIPHER_BLOCK_LENGTH(key_type),          \
                               (input_length) + 1) +                             \
-     PSA_CIPHER_IV_LENGTH((key_type), (alg)) :                                  \
+     PSA_CIPHER_IV_LENGTH((key_type), (alg)) : 0) :                             \
      (PSA_ALG_IS_CIPHER(alg) ?                                                  \
       (input_length) + PSA_CIPHER_IV_LENGTH((key_type), (alg)) :                \
      0))
```
Comment (gilles-peskine-arm) [inline include/psa/crypto_sizes.h:1002]: `PSA_CIPHER_UPDATE_OUTPUT_SIZE` has a similar issue.

File: include/psa/crypto_sizes.h [hunk 2]
```diff
@@ -992,9 +992,11 @@
  */
 #define PSA_CIPHER_ENCRYPT_OUTPUT_SIZE(key_type, alg, input_length)             \
     (alg == PSA_ALG_CBC_PKCS7 ?                                                 \
+     (((key_type) & PSA_KEY_TYPE_CATEGORY_MASK)                                 \
+                                           == PSA_KEY_TYPE_CATEGORY_SYMMETRIC ? \
```
Comment (gilles-peskine-arm) [inline include/psa/crypto_sizes.h:996]: While this is correct, `PSA_BLOCK_CIPHER_BLOCK_LENGTH(key_type)` would make the correctness more obvious.

**Review by gilles-peskine-arm — CHANGES_REQUESTED** [review]
Please add a non-regression test and a changelog entry.

**Review by TRodziewicz — APPROVED** [review]
LGTM.

**Review by gilles-peskine-arm — APPROVED** [review]

**Pull request merged**